src/mmrelay/matrix_utils.py

```
@@ -2955,125 +3269,169 @@ async def on_room_message(
+        response, _ = await client.upload(
+            io.BytesIO(image_data),
+            content_type=content_type,
+            filename=filename,
+            filesize=len(image_data),
+        )
+    except NIO_COMM_EXCEPTIONS as e:
+        # Convert nio communication exceptions to an UploadError-like instance
+        logger.exception("Image upload failed due to a network error")
+        try:
+            upload_error = UploadError(message=str(e), status_code=None)
+        except Exception:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)